crates/server/src/authority/authority.rs

```diff
@@ -227,6 +227,13 @@ pub trait Authority: Send + Sync {
     /// Returns the kind of non-existence proof used for this zone.
     #[cfg(feature = "__dnssec")]
     fn nx_proof_kind(&self) -> Option<&NxProofKind>;
+
+    /// Returns the store label used within metrics.
+    /// In case the label is not present metrics may not be generated for the authority.
+    #[cfg(feature = "metrics")]
+    fn metrics_store_label(&self) -> Option<&'static str> {
```

Comment:
Why does this yield an `Option`? I think it might be better represented as an associated const.

Also don't love the name... we use "store" in the server crate in a way that's somewhat independent of the `Authority` trait, which is is a part of. How about just `metrics_label()`?

Reply:
The thought behind the `Option` was that for example the `InMemoryAuthority` is used within other Authorities.
In terms of metrics, this could lead to situations where counts could be duplicated. Therefore `Option` seemed reasonable to be able to skip metrics for Authorities. It as well allowed to stay backwards compatible.

Changing the name to `metrics_label` is fine, in case it is clear that it will be a single label per authority. In this case I'd rename the label on the metric to `authority`.

Reply:
The associated const. approach does not work out:
```
   |  ^^^^^^^^^^^^^ `Authority` is not dyn compatible
79 |     const METRICS_LABEL : &'static str;
   |           ^^^^^^^^^^^^^ ...because it contains this associated `const`

Reply:
> The associated const. approach does not work out:
> 
> ```
>    |  ^^^^^^^^^^^^^ `Authority` is not dyn compatible
> ...
> 79 |     const METRICS_LABEL : &'static str;
>    |           ^^^^^^^^^^^^^ ...because it contains this associated `const`
> ```

Ah, okay, that makes sense -- sorry for the misleading suggestion.

Reply:
Changed authority trait fn to: `fn metrics_label(&self) -> &'static str;`
Added metrics label for all authorities.
Renamed metric label on `hickory_zone_lookups_total` from `store` to `authority`.

Reply:
> > The thought behind the `Option` was that for example the `InMemoryAuthority` is used within other Authorities. In terms of metrics, this could lead to situations where counts could be duplicated. Therefore `Option` seemed reasonable to be able to skip metrics for Authorities. It as well allowed to stay backwards compatible.
> 
> This makes sense but I don't think it's clearly reflected in the documentation:

At the moment there is no such case where it does affects the results, so I've meanwhile removed the `Option`.
I'm not sure if there'll be a case where this thought/nesting will apply and have a real effect on the metric counters, so I'm unsure which variant is the better approach.